prepline_general/api/general.py

```diff
@@ -278,13 +278,16 @@ def pipeline_api(
 
             # This will raise if the file is encrypted
             pdf.metadata
-        except pypdf.errors.EmptyFileError:
+        except (pypdf.errors.EmptyFileError, pypdf.errors.PdfStreamError):
             raise HTTPException(status_code=400, detail="File does not appear to be a valid PDF")
         except pypdf.errors.FileNotDecryptedError:
             raise HTTPException(
                 status_code=400,
                 detail="File is encrypted. Please decrypt it with password.",
             )
+        except Exception as e:
+            logger.warn(f"PDF read error: {e}")
+            raise HTTPException(status_code=400, detail="File does not appear to be a valid PDF")
```

Comment:
can we combine the two `HTTPException(status_code=400, detail="File does not appear to be a valid PDF")`? (another one on line 282)

Reply:
I added the generic catch to log whatever we just caught. This will let us track anything else that's getting thrown, and decide how to handle it.

Reply:
Adjusted this per Crag's note below